Source/DafnyCore/Compilers/Python/PythonBackend.cs

```diff
@@ -46,6 +47,17 @@ public class PythonBackend : ExecutableBackend {
     return Path.GetExtension(externFilename) == ".py" ? Path.GetFileNameWithoutExtension(externFilename) : null;
   }
 
+  private static readonly Dictionary<OSPlatform, string> PlatformDefaults = new() {
+    {OSPlatform.Linux, "python3"},
+    {OSPlatform.Windows, "python"},
+    {OSPlatform.FreeBSD, "python3"},
+    {OSPlatform.OSX, "python3"},
+  };
+  private static string DefaultPythonCommand => PlatformDefaults.SingleOrDefault(
```

Comment:
I think you could have done `PlatformDefaults.SingleOrDefault(kv => RuntimeInformation.IsOSPlatform(kv.Key)) ?? "python3"` as well, to drop the unused `OSPlatform.Linux` value

Reply:
Good catch. I was spinning my wheels on this for a few minutes lol. Your suggestion maybe isn't completely correct because the result of the single call would end up being a KV pair if it succeeded, I think. So I think the full "gudder" way would have been `blah.SingleOrDefault(P)?.Value ?? "python3"`. Is this correct? I usually use Python at work so I don't get a lot of C# road time. IIRC `.?` is the null coalescing accessor which would propogate the null of the `SingleOrDefault`, then the `??` would drop in the `"python3"` if the LHS is null. Please correct me if I'm wrong, like I said, I primarily do Python.

P. S. Apologies on laziness on the shortened parts, I'm on my phone.

Reply:
Yes, you might be right !